internal/resource/url.go

```diff
@@ -352,9 +352,19 @@ func (f *Fetcher) FetchFromS3(u url.URL, dest *os.File, opts FetchOptions) error
 
 	sess.Config.Region = aws.String(region)
 
+	var versionId *string
+	if v, ok := u.Query()["versionId"]; ok && len(v) > 0 {
+		if v[0] == "" {
```

Comment:
This can be handled at validation time and should be moved there. I'm on the fence whether it should be a fatal error or not.

Reply:
Let's make it a hard failure, so that it's easier to spot when a mistake happens on user or transpiler side.

Reply:
Sounds good to me. Since I'm new to this codebase, I'm planning on adding this validation here: https://github.com/coreos/ignition/blob/master/config/v2_2/types/url.go#L36, this is the right place, right? Should I also add this to v2.1, since we support s3 in there also?

Reply:
Yup! You'll want to add it to v2_1, v2_2, and v2_3_experimental.